torch/_numpy/_reductions_impl.py

```diff
@@ -421,9 +421,12 @@ def percentile(
     *,
     interpolation: NotImplementedType = None,
 ):
+    dt = _dtypes_impl.default_dtypes().float_dtype
+    qq = q / torch.as_tensor(100.0, dtype=dt)
```

Comment:
How is this preferable?

Reply:
Int tensor / 100.0 is float32 otherwise. Found via an assert_equal in TestMedian failing with abs errors of the order of 1e-7.

Reply:
Ugh, right. In order to keep the scalars being scalar for now, rather than doing this, let's do a cast for `q` when `q` is integer.

Wrapping scalars into tensors is best avoided for now, as it deactivates a number of optimisations within PyTorch.

Reply:
`q` is normalized as ArrayLike already, so it's a tensor at this point even if the original input was a scalar.

Reply:
I mean wrapping 100.0 into a tensor

Reply:
OK, done in https://github.com/pytorch/pytorch/pull/112959/commits/ff290b04fa1f58eb1efdd435f41317d98569d25f

Reply:
The current implementation still wraps 100. I meant
```python
    if _dtypes_impl.python_type_for_torch(q.dtype) == int:
        q = q.to(_dtypes_impl.default_dtypes().float_dtype)
    qq = q / 100.0
```

Reply:
OK, done.